### ISSUE TITLE
[Solidity] How to initialize multi-dimensional dynamic storage arrays

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#8018]
## Description

A 2D dynamic storage array such as `bool[][] sv_0` used to be initialized by writing to the length member of the array and initializing the slots.

```
contract C {
   bool[][] sv_0;
   function foo() public {
      sv_0.length = 1;
      sv_0[0].length = 2;
      sv_0[0][0] = true;
      sv_0[0][0] = false;
  }
}
```

Now that writing to the `.length` field is prohibited (see https://github.com/ethereum/solidity/pull/7350), how does one do it now using `push`?

**Comment by bshastry** [comment]
I suppose one way to do it would be like so

```
bool[] sv_1;
sv_1.push(true);
sv_1.push(false);
sv_0.push(sv_1);
```

Are there other alternatives that one could consider?

**Comment by chriseth** [comment]
There is also the new push member without parameters:

``for (uint i = 0; i < length; i ++) sv_1.push();```

It pushes a zero-initialized element and is more or less identical to `length++`.

**Comment by bshastry** [comment]
Closed by #8004
